adapters/repos/db/index.go

```diff
@@ -110,7 +124,19 @@ func (m *shardMap) RangeConcurrently(logger logrus.FieldLogger, f func(name stri
 	eg := enterrors.NewErrorGroupWrapper(logger)
 	eg.SetLimit(_NUMCPU)
 	(*sync.Map)(m).Range(func(key, value any) bool {
-		name, shard := key.(string), value.(ShardLike)
+		name, ok := key.(string)
```

Comment:
You can make this much shorter by adding a defer recover(), which will catch all the panics, not just the type conversions